### ISSUE TITLE
Command lines like `x=1` set $status to 127 but report no error

### ISSUE THREAD

**Issue by msteed** [fish-shell/fish-shell#4338]
```
> echo $version
2.6.0
> uname -a
Linux strongbad 4.12.8-1-ARCH #1 SMP PREEMPT Thu Aug 17 08:37:20 CEST 2017 x86_64 GNU/Linux
```

Trying to run an uniknown command that contains `=` sets `$status` to 127 but does not print an error message:
```
> sh -c 'env HOME=$(mktemp -d) fish'
Welcome to fish, the friendly interactive shell
> x=1
> echo $status
127
> =1
> echo $status
127
> x=
> echo $status
127
> =
> echo $status
127
```

For unknown commands that do not contain `=`, fish reports an error:
```
> x1
fish: Unknown command 'x1'
```

**Comment by faho** [comment]
I'm quite certain we already have an issue open about this, but I can't find it.

The idea here is that this "command" gets a different error message because we assume that someone is trying to do POSIX-style variable assignment.

It appears that code has a bug.

**Comment by krader1961** [comment]
This  has already been fixed in the upcoming 2.7.0 release. Several attempts to fix how errors get reported resulted in one problem being fixed while introducing a different problem. We think that commit 5e2c14e7 to fix #4221 has corrected all the issues.
